### PR TITLE
[move-compiler] Improve error message for if without else

### PR DESCRIPTION
**Pull request by tnowacki**
## Description 

- Improve error message for if without an else
- Persisting through typing for lints

## Test plan 

- New test

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

File: external-crates/move/crates/move-compiler/src/typing/ast.rs [hunk 1]
```diff
@@ -191,7 +191,7 @@ pub enum UnannotatedExp_ {
     Builtin(Box<BuiltinFunction>, Box<Exp>),
     Vector(Loc, usize, Box<Type>, Box<Exp>),
 
-    IfElse(Box<Exp>, Box<Exp>, Box<Exp>),
+    IfElse(Box<Exp>, Box<Exp>, Option<Box<Exp>>),
```
Comment (tnowacki) [inline external-crates/move/crates/move-compiler/src/typing/ast.rs:194]: We don't need this in typing just for the error message, but I am persisting it for lints

File: external-crates/move/crates/move-compiler/src/typing/translate.rs [hunk 2]
```diff
@@ -1586,15 +1588,28 @@ fn exp(context: &mut Context, ne: Box<N::Exp>) -> Box<T::Exp> {
...
+                    || "Incompatible branches",
+                    et.ty.clone(),
+                    ef.ty.clone(),
+                ),
+                None => {
+                    let ty = sp(eloc, Type_::Unit);
+                    subtype(
+                        context,
+                        eloc,
+                        || "Invalid 'if'. The body of an 'if' without an 'else' must be type '()'",
```
Comment (tzakian) [inline external-crates/move/crates/move-compiler/src/typing/translate.rs:1605]: Very nitty nit: 
```suggestion
                        || "Invalid 'if'. The body of an 'if' without an 'else' must be of type '()'",
```

Comment (tzakian) [reply external-crates/move/crates/move-compiler/src/typing/translate.rs:1605]: or `... must have type '()'`

**Review by tzakian — APPROVED** [review]
LGTM. Small nit on error message wording.

**Pull request merged**